src/services/tx/sponsoredCall.ts

```diff
@@ -18,14 +28,30 @@ export const sponsoredCall = async (tx: SponsoredCallPayload): Promise<{ taskId:
     body: JSON.stringify(tx),
   }
 
-  const data = await fetch(SAFE_GELATO_RELAY_SERVICE_URL, requestObject).then((res) => {
+  const res = await fetch(SAFE_GELATO_RELAY_SERVICE_URL, requestObject)
+
+  if (res.ok) {
+    return res.json()
+  }
+
+  const errorData: { error?: { message: string } } = await res.json()
+  throw new Error(`${res.status} - ${res.statusText}: ${errorData?.error?.message}`)
+}
+
+export const getRemainingRelays = async (chainId: string, address: string): Promise<number> => {
```

Comment:
Nit: now this is here, can we rename this file to `relaying.ts` or something similar?